packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -396,6 +396,14 @@ export class Table extends Construct {
       errors.push('a sort key of the table must be specified to add local secondary indexes');
     }
 
+    // reshape GSI and LSI since they can be added in a lazy manner
+    if (this.globalSecondaryIndexes.length === 0) {
```

Comment:
I'd rather you change the location where we pass these to the L1 into something like this:

```ts
globalSecondaryIndexes: new cdk.Token(() => this.globalSecondaryIndexes.length > 0 ? this.globalSecondaryIndexes : undefined),
```

Reply:
Thanks for a valuable comment! Let me update to use `cdk.Token`.